### ISSUE TITLE
ALT text entered in the compose dialog is erased when you press "Edit" again

### ISSUE THREAD

**Issue by mackuba** [bluesky-social/social-app#996]
**Describe the bug**

When you're composing a new post, you enter some ALT text and then you click the "Edit" (pencil) button again, the ALT text is missing. If you press Save, it's deleted from the composed post and lost - which may be really annoying if you have already entered a several sentences long alt text and you need to retype it again.

**To Reproduce**

1. Click "New post".
2. Click the image button and upload an image.
3. Press the ALT or the Edit (pencil) button.
4. Enter an alt text and press Save. (you're back in the main compose window)
5. Press the Edit (pencil) button again.
  - The ALT text is now empty
6. If you press Cancel -> you're back in the main compose window with ALT still set.
7. If you press Save -> you're back in the main compose window, but the ALT text has been erased.

**Expected behavior**

The ALT text you've previously composed should be visible in the ALT text field when you reopen the Edit screen again.

**Screenshots**

I upload an image and enter and save an ALT text:

<p><img width="594" alt="Screen Shot 2023-07-07 at 20 00 59" src="https://github.com/bluesky-social/social-app/assets/28465/9cf1eb76-721b-4d33-b172-c220050a88a5"></p>

The ALT text is set (green checkmark):

<p><img width="663" alt="Screen Shot 2023-07-07 at 20 00 36" src="https://github.com/bluesky-social/social-app/assets/28465/815dd840-7a70-49f3-b5be-f90aeb84a12b"></p>

When I click Edit (pencil icon) again, the ALT text is gone:

<p><img width="651" alt="Screen Shot 2023-07-07 at 20 01 09" src="https://github.com/bluesky-social/social-app/assets/28465/f3aaea6d-4245-4527-8c08-7c59ba70b252"></p>

**Details**

 - Platform: macOS 12.6.6, Safari 16.5.1

**Comment by hkolbeck** [comment]
I'm hitting this too. macOS 12.6.6, Chrome 114.0.5735.198


https://github.com/bluesky-social/social-app/assets/225933/62290b88-d015-4b23-b646-8b131909272b

**Comment by pfrazee** [comment]
Ah, my apologies. I had received this report and couldn't reproduce -- I had forgotten Web has an additional modal for setting the alt text. Thanks for opening this issue and helping me suss out what I missed.

**Comment by jaysoni-ash42** [comment]
@pfrazee i have made the local change which looks like this if this resolves this issue can i pull up a pr

https://github.com/bluesky-social/social-app/assets/68722497/555db251-ce85-4a81-b7d3-4afe09f60070

**Comment by jaysoni-ash42** [comment]
@pfrazee  i have made some changes locally which looks like this if it resolves this issue should i pull up a pr for it.

https://github.com/bluesky-social/social-app/assets/68722497/7f9e96be-b39e-4d27-8f1e-6f51fc700a8d

**Comment by pfrazee** [comment]
Appreciate it!
